src/Sylius/Bundle/ApiBundle/Applicator/StateMachineTransitionApplicator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Applicator;
+
+use SM\Factory\FactoryInterface as StateMachineFactoryInterface;
+use Sylius\Component\Core\Model\ProductReview;
+use Sylius\Component\Core\ProductReviewTransitions;
+use Sylius\Component\Review\Model\ReviewInterface;
+
+final class StateMachineTransitionApplicator
+{
+    /** @var StateMachineFactoryInterface $stateMachineFactory */
+    private $stateMachineFactory;
+
+    public function __construct(StateMachineFactoryInterface $stateMachineFactory)
+    {
+        $this->stateMachineFactory = $stateMachineFactory;
+    }
+
+    public function accept(ProductReview $data): ReviewInterface
```

Comment:
I don't remember the results, can you remind me what will happen if we will use an interface here?
```suggestion
    public function accept(ReviewInterface $data): ReviewInterface
```

Reply:
the same result as using class, I will change it to interface